packages/components/src/hooks/use-deep-effect.js

```diff
@@ -0,0 +1,11 @@
+import * as React from 'react';
+import { isDeepEqual } from '@deriv/shared';
+
+export const useDeepEffect = (callback, dependencies) => {
+    const prev_dependencies = React.useRef(null);
+
+    if (!isDeepEqual(prev_dependencies, dependencies)) {
```

Comment:
How do we make sure that this is not applying on objects with circular refs?

Reply:
Probably you shouldn't use this effect if it has any circular references, i.e. don't use this effect on huge objects cause it's quite a pricey call to make on each re-render. I'll add a comment to explain.